ocaml/xapi/xapi_vlan.ml

```diff
@@ -43,13 +43,38 @@ let create_internal ~__context ~host ~tagged_PIF ~tag ~network ~device =
   let () = Db.VLAN.create ~__context ~ref:vlan ~uuid:vlan_uuid ~tagged_PIF ~untagged_PIF ~tag ~other_config:[] in
   vlan, untagged_PIF
 
+let assert_vlan_on_pif_compatible_with_network ~__context candidate_pif candidate_pif_topo network =
+  let open Xapi_pif_helpers in
+  match Db.Network.get_PIFs ~__context ~self:network with
+  | existing_pif :: _ ->
```

Comment:
Simplify the code (also double check if you still need the `begin`/`end`, I think not):
```ocaml
match Db.Network.get_PIFs ~__context ~self:network with
| [] -> ()
| pif :: _ ->
    let pif_rec = Db.PIF.get_record ~__context ~self:pif in
    let pif_topo = get_pif_topo ~__context ~pif_rec in
    match pif_topo, candidate_pif_topo with
    | VLAN_untagged _ :: Network_sriov_logical existing_sriov :: _
    , Network_sriov_logical candidate_sriov :: _ ->
        (* both based on sriov logical, then compare sriov compatibility *)
        let existing = Db.Network_sriov.get_physical_PIF ~__context ~self:existing_sriov in
        let candidate = Db.Network_sriov.get_physical_PIF ~__context ~self:candidate_sriov in
        if not (is_device_underneath_same_type ~__context existing candidate) then
          raise Api_errors.(Server_error (vlan_network_incompatible, [Ref.string_of candidate; Ref.string_of network]))
    | VLAN_untagged _ :: Network_sriov_logical _ :: _, _
        (* existing pif based on sriov logical but candidate pif does not *)
    | VLAN_untagged _ :: _, Network_sriov_logical _ :: _ ->
        (* existing pif not based on sriov logical but candidate pif does *)
        raise Api_errors.(Server_error (vlan_network_incompatible, [Ref.string_of candidate; Ref.string_of network]))
    | _ -> ()
```

Reply:
I cannot find out how to simply the code anymore except remove the `begin` and `end`.